### PR TITLE
[ENH] Dockerize the query tool

### LINKED ISSUE #183: Provide local static deploy option that is containerized
Query tool should be dockerized to also have a local deployment that interfaces with the existing docker compose up API deployment

  - [x] choose a process to generate a statically rendered page for the query tool
  - [x] choose and set up some kind of http server that can serve the static page locally
  - [x] create a Dockerfile that defines the runtime env for the query tool during static file generation
  - [x] make sure to use npm install that targets the package-lock.json to have deterministic builds
  - [x] consider using https://docs.docker.com/build/building/multi-stage/ to reduce the size of the built image (optional - maybe later)
  - [x] configure a Docker entrypoint that launches the http server and serves the generated static site contents via some port - and make sure that port is exposed: https://docs.docker.com/engine/reference/builder/#expose
  - [x] update https://github.com/neurobagel/query-tool#launch-the-query-tool to explain how to use the new Docker run workflow

As a user, I want to be able to make a query tool run locally on my machine without installing anything beyond docker, so that I can have it query my already locally deployed neurobagel API stack.

Thinking:
- nuxt generate to build static page
- npm start for very basic npm http server
- npm install directly from the package-lock.json thing
- as long as the deployment target is set to "static": https://v2.nuxt.com/docs/features/deployment-targets/

### LINKED ISSUE #195: Query form can't access `API_QUERY_URL` when passed in as a shell environment variable
While attempting to pass in a value for `API_QUERY_URL` to the query tool in a Docker container, we discovered that we cannot provide a basic shell environment variable to the query tool that appears on the client side using the dotenv module. In the current query tool setup, the only way for an env variable to be visible is if it's defined in the `.env` file (and in the case of a Docker container, made accessible in container as a file, e.g., by mounting it or copy everything over).

This is due to shell variables + `.env` file variables being treated very differently by Nuxt itself. To make shell variables visible on the client side in a non-security-risky way (and also to enable us to more easily configure things at runtime in Docker), we need to:
- [x] update `nuxt.config.js` to define `API_QUERY_URL` in the `publicRuntimeConfig` object

Resources:
- https://v2.nuxt.com/docs/configuration-glossary/configuration-env/
- https://v2.nuxt.com/docs/configuration-glossary/configuration-runtime-config#publicruntimeconfig
- https://stackoverflow.com/questions/67703133/how-to-use-env-variables-in-nuxt-2-or-3
- https://v2.nuxt.com/tutorials/moving-from-nuxtjs-dotenv-to-runtime-config/#migrating-to-the-nuxtjs-runtime-config-from-nuxtjsdotenv

### PR DESCRIPTION
**Pull request by surchs**
Closes #183
Closes #195 

Changes proposed in this pull request:

- Add Dockerfile to define dependencies
  - Explicitly define ENV variables for Nuxt port and host IP that can be overridden at runtime 
- Update `nuxt.config.js` to define variable to be client-accessible using `publicRuntimeConfig`
- basic README updates (probably good to test during review)
- also add Docker deploy to deploy workflow

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,7 +94,20 @@ if you're using the remote api, your `.env` file would look something like this:
 API_QUERY_URL=https://api.neurobagel.org/
 ```
 
-### Launch the Query Tool
+
+### Launch the Query Tool as a Docker container
+To launch the query tool Docker container, simply run
+
+```bash
+docker run -p 8000:80 neurobagel/query_tool:latest
```
Comment (alyssadai) [inline README.md]:
```suggestion
docker run -p 8000:3000 --env-file=.env neurobagel/query_tool:latest
```

File: README.md [hunk 2]
```diff
@@ -81,7 +94,20 @@ if you're using the remote api, your `.env` file would look something like this:
 API_QUERY_URL=https://api.neurobagel.org/
 ```
 
-### Launch the Query Tool
+
+### Launch the Query Tool as a Docker container
+To launch the query tool Docker container, simply run
```
Comment (alyssadai) [inline README.md]:
```suggestion
To launch the query tool Docker container and pass in the `.env` file you have created, simply run
```

**Review by alyssadai — APPROVED** [review]
Looks good @surchs!

Just a few suggestions to align the README with the changes we've made, otherwise 🧑‍🍳

**Comment by surchs** [comment]
Also adding https://github.com/neurobagel/query-tool/issues/196 to create some testable behaviour when API_QUERY_URL is not defined on the client side

**Pull request merged**